source/agora/consensus/protocol/Nominator.d

```diff
@@ -512,6 +518,21 @@ extern(D):
                     envelope.statement.pledges.confirm_.ballot.value, ex);
                 return;
             }
+
+            // If it's an old envelope, we're only interested in the signature
+            if (envelope.statement.slotIndex <= last_block.header.height)
+            {
+                const sig = ValidatorBlockSig(
+                    Height(envelope.statement.slotIndex), public_key,
+                    Scalar(envelope.statement.pledges.confirm_.value_sig));
+                auto blocks = this.ledger.getBlocksFrom(Height(envelope.statement.slotIndex));
+                if (!this.collectBlockSignature(sig, blocks[0].hashFull()))
+                    log.trace("Added signature for {} from CONFIRM ballot", public_key);
```

Comment:
Is this the right message?

Reply:
I think so, what do you see as problematic ?

Reply:
It returns `false`, which means `Invalid point from public key` or `INVALID Block signature received`.

Reply:
🤦 
You're right, I'll fix